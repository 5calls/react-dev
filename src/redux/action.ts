import { Action } from 'redux';
import { Issue } from '../common/model';

export enum ActionType {
  GET_ISSUES,
  ISSUE_SELECTED,
}

export interface IssuesAction extends Action {
  type: ActionType;
  payload: Issue[];
}

export interface IssueSelectedAction extends Action {
  type: ActionType;
  payload: Issue;
}
