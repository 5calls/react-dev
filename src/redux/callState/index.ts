export {
  CallStateAction, CallStateActionType,
  CurrentIssueAction, NextContact,
  SetContactIdsAction, SetShowFieldOfficeNumbers,
  CompleteIssueAction, ClearContactIndexesAction,
  JoinGroupAction } from './action';
export {completeIssueActionCreator, moveToNextActionCreator,
  selectIssueActionCreator, clearContactIndexes, joinGroupActionCreator } from './actionCreator';
export { callStateReducer, CallState } from './reducer';
export { OutcomeData, FlexibleOutcomeData, submitOutcome, submitFlexibleOutcome } from './asyncActionCreator';
