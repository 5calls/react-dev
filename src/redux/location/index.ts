export { LocationAction, LocationActionType, LocationClearedAction,
  LocationSetAction, InvalidAddressAction, SetUiStateAction,
  CacheCityAction, FetchingLocationAction, ValidatingLocationAction } from './action';
export { setLocation, clearAddress, setInvalidAddress, setCachedCity,
  setFetchingLocation, setValidatingLocation, setUiState} from './actionCreator';
export { LocationState, locationStateReducer } from './reducer';
export { setAddress } from './asyncActionCreator';
