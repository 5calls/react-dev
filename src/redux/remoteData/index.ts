export { IssuesAction, RemoteDataAction, RemoteDataActionType, CallCountAction } from './action';
export { issuesActionCreator, callCountActionCreator } from './actionCreator';
export { RemoteDataState, remoteDataReducer } from './reducer';
export { getIssues, fetchCallCount } from './asyncActionCreator';
