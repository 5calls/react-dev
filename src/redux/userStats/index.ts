export {
  UserStatsAction, UserStatsActionType, SetUserStatsAction, AddCallEventAction
} from './action';
export { setUserStatsActionCreator, addCallEventActionCreator } from './actionCreator';
// export { updateUserStatsAsyncActionCreator } from './asyncActionCreator';
export { userStatsReducer, UserStatsState, UserContactEvent } from './reducer';
