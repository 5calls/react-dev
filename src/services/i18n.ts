import * as i18n from 'i18next';
import * as XHR from 'i18next-xhr-backend';
// import Cache from 'i18next-localstorage-cache';
import * as LanguageDetector from 'i18next-browser-languagedetector';

const instance = i18n
    .use(XHR)
    // .use(Cache)
    .use(LanguageDetector)
    .init({
        backend: {
            loadPath: '/locales/{{lng}}.json'
        },
        fallbackLng: 'en',

        react: {
            wait: true, // globally set to wait for loaded translations in translate hoc
            // exposeNamespace: true // exposes namespace on data-i18next-options to be used in eg. locize-editor
        },

        // have a common namespace used around the full app
        ns: [''],
        defaultNS: '',

        debug: true,

        // cache: {
        //   enabled: true
        // },

        interpolation: {
            escapeValue: false, // not needed for react!!
            formatSeparator: ','
        }
    });

export default instance;
