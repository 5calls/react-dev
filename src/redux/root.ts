import { ApplicationState } from './root';
import { routerReducer, RouterState } from 'react-router-redux';
import { Reducer, combineReducers } from 'redux';
import { LocationState, locationStateReducer } from './location/reducer';
import { CallState, callStateReducer } from './callState/reducer';
import { RemoteDataState, remoteDataReducer } from './remoteData/reducer';

export interface ApplicationState {
  remoteDataState: RemoteDataState;
  callState: CallState;
  locationState: LocationState;
}

export const DefaultApplicationState: ApplicationState = {
  remoteDataState: {} as RemoteDataState,
  callState: {} as CallState,
  locationState: {} as LocationState
};

// DANGER: TypeScript magic ahead!!
// These articles should help your understanding of this section:
// https://blog.mariusschulz.com/2017/01/06/typescript-2-1-keyof-and-lookup-types
// https://blog.mariusschulz.com/2017/01/20/typescript-2-1-mapped-types
// https://moin.world/2017/06/18/10-TypeScript-Features-You-Might-Not-Know/
//
// This will be used by redux-persist in store.ts
// A type that is an ApplicationState key
export type ApplicationStateKeyType = keyof ApplicationState;
// A TS mapped type to type out an object shape that represents
// all ApplicationState keys and their string equivalents.
type ApplicationStateKeyTypes = { [K in ApplicationStateKeyType]: ApplicationStateKeyType };
// Object that maps out all possible ApplicationState keys
export const ApplicationStateKey: ApplicationStateKeyTypes = {
  locationState: 'locationState',
  remoteDataState: 'remoteDataState',
  callState: 'callState'
};

const rootReducer = combineReducers({
  routing: routerReducer as Reducer<RouterState>,
  remoteDataState: remoteDataReducer,
  callState: callStateReducer,
  locationState: locationStateReducer
});

export default rootReducer;
