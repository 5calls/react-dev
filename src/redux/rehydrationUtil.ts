import { Dispatch } from 'redux';
import { ApplicationState } from './root';

// when things are dispatched on component mount on first page load, 
// it's possible that this can occur before rehydration is complete.
// keep track of this and launch any post-hydration actions afterwards

let hasRehydrated = false;
let rehydrationQueue: Function[] = [];

export const onStorageRehydrated = () => {
  return (dispatch: Dispatch<ApplicationState>,
          getState: () => ApplicationState) => {
    // console.log(`rehydrated, executing`);

    hasRehydrated = true;
    
    for (let f of rehydrationQueue) {
      f();
    }
  
    rehydrationQueue = [];
  };
};

export const queueUntilRehydration = (f: Function) => {
  if (hasRehydrated) {
    // console.log(`already rehydrated, executing`);
    f();
  } else {
    // console.log(`storing for later`);
    rehydrationQueue.push(f);
  }
};