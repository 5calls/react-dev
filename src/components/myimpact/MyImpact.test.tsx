import * as React from 'react';
import { shallow } from 'enzyme';
import i18n from '../../services/i18n';
import { MyImpact } from './index';
import { UserStatsState } from '../../redux/userStats';

test('Done component snapshot renders correctly', () => {
  const userStats: UserStatsState = {
    all: [],
    voice_mail: 0,
    unavailable: 0,
    made_contact: 0,
    yes: 0,
  };
  const count = 1000;
  const component = shallow(
    <MyImpact
      userStats={userStats}
      totalCount={count}
      t={i18n.t}
    />
  );
  expect(component).toMatchSnapshot();
});
