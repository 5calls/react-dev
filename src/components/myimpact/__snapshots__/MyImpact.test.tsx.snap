// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyImpact component snapshot renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="impact"
>
    <h2
        className="impact__title"
    />
    <h2
        className="impact_total"
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <p
        className="impact__text"
    />
    <div
        className="impact_result"
        dangerouslySetInnerHTML={
            Object {
                "__html": undefined,
              }
        }
    />
    <Translate(Component)
        t={[Function]}
        totalCount={1000}
    />
</section>,
  "nodes": Array [
    <section
      className="impact"
>
      <h2
            className="impact__title"
      />
      <h2
            className="impact_total"
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <p
            className="impact__text"
      />
      <div
            className="impact_result"
            dangerouslySetInnerHTML={
                  Object {
                        "__html": undefined,
                      }
            }
      />
      <Translate(Component)
            t={[Function]}
            totalCount={1000}
      />
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        t={[Function]}
        totalCount={1000}
        userStats={
                Object {
                        "all": Array [],
                        "made_contact": 0,
                        "unavailable": 0,
                        "voice_mail": 0,
                        "yes": 0,
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "t": [Function],
          "totalCount": 1000,
          "userStats": Object {
            "all": Array [],
            "made_contact": 0,
            "unavailable": 0,
            "voice_mail": 0,
            "yes": 0,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="impact"
>
          <h2
                    className="impact__title"
          />
          <h2
                    className="impact_total"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    className="impact__text"
          />
          <div
                    className="impact_result"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <Translate(Component)
                    t={[Function]}
                    totalCount={1000}
          />
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="impact"
>
          <h2
                    className="impact__title"
          />
          <h2
                    className="impact_total"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <p
                    className="impact__text"
          />
          <div
                    className="impact_result"
                    dangerouslySetInnerHTML={
                              Object {
                                        "__html": undefined,
                                      }
                    }
          />
          <Translate(Component)
                    t={[Function]}
                    totalCount={1000}
          />
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    t={[Function]}
    totalCount={1000}
    userStats={
        Object {
            "all": Array [],
            "made_contact": 0,
            "unavailable": 0,
            "voice_mail": 0,
            "yes": 0,
          }
    }
/>,
}
`;
